### DISCUSSION TITLE
Possible to Read/Set Review Status and Checkmark Properties on Text Annotation?

### DISCUSSION THREAD

**Discussion by JiongQ** [pymupdf/PyMuPDF#1984] [Q&A]
May I know is it possible to read/set review status (none, accepted, rejected, canceled) and checkmark properties (shown in the comment pane of adobe pdf reader when a highlight is selected) of a text annotation?

I believe review status and checkmark property should be one of the metadata bound to annotation, I can please have advice on the steps to get or set these properties using pymupdf?  

![Check Mark Status](https://user-images.githubusercontent.com/74604784/197374041-f1cdc608-5c42-40b8-96a5-669bf54c7e11.JPG)
![Review Status](https://user-images.githubusercontent.com/74604784/197374044-a17bd4e4-465f-4f94-8b4a-a13e0a6fc6fb.JPG)

**Comment by JorjMcKie** [comment]
This is currently not supported. And it neither is supported by most PDF viewers apart from Adobe itself.
Implemented like this:
* add a number of **_hidden_** (never visible) "Text" annotations (like sticky notes) to the page
* each of these notes ('Text' annotations) are flagged with the PDF property `/IRT` (= "in response to") that points to the highlight annotation
* certain properties of each note then reflect e.g. the `/State` (like checkmarked or accepted).

Using low-level methods of PyMuPDF, you can find your way to these notes and check the mentioned properties.
I do not believe we will provide a high-level interface to this, but you can build your own code to do this - drop me a note if you are interested.
